.github/workflows/00_pull_request_entry_point.yml

```diff
@@ -0,0 +1,50 @@
+name: Pull Request Entry Point
+
+on:
+  pull_request:
+
+jobs:
+  check-permissions:
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'run-no-ci') }}
+    uses: ./.github/workflows/90_helpers_check-permissions.yml
+    with:
+      github-event-name: ${{ github.event_name }}
+
+  projects-changed:
+    needs: [ check-permissions ]
+    name: Check source files for changes
+    runs-on: ubuntu-22.04
+    timeout-minutes: 2
+    outputs:
+      go-projects-changes: ${{ steps.go-projects-changes.outputs.changes }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Go Projects Changes
+        id: go-projects-changes
+        uses: dorny/paths-filter@1441771bbfdd59dcd748680ee64ebd8faab1a242  #v3
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters/go-build.yaml
+          base: ${{ github.event_name != 'pull_request' && (github.event.merge_group.base_ref || github.ref_name) || '' }}
+          ref: ${{ github.event_name != 'pull_request' && (github.event.merge_group.head_ref || github.ref) || '' }}
+
+  build-go-projects:
+    needs: [ check-permissions, projects-changed ]
+    if: ${{ needs.projects-changed.outputs.go-projects-changes != '[]' }}
+    strategy:
+      matrix:
+        cmd-path: ${{ fromJson(needs.projects-changed.outputs.go-projects-changes) }}
+    uses: ./.github/workflows/50_build_go_cmd.yml
```

Comment:
regarding the file names I have two questions:
 - why the numbered prefix?
 - in other repos we've had a mix of `snake_case` and `kebap-case` for the file names. For most things, we seem to be moving towards `kebap-case`, any specific reason we're going `snake_case` here?

Reply:
good catch, thank you!

for the numbers: that will keep order of the workflows in the folder for better structure. Kind of `00 - 10` entry points, `90 - 99` are "helpers" with small functionality and main actions also shoud be grouped near, so thats how I find easier to navigate and have `build` and `tests` files near.

in other repos we also use underscore for helpers, just having numbers allow to keep order and structure more flexible.

and I also tried options to introduce folders for the structure, only too hacky ways for doing that, and by default GH expect all workflow files to be in the one folder :( 

as for cases: I could use underscore to separate leading number from the text, and for the text I will use kebap-case